### PR TITLE
Updating polars and sqlparser versions

### PR DESCRIPTION
**Pull request by ayax79**
Polars and SQLParser upgrade.

I have exposed features that have been added to polars as command args where appropriate.

### CODE REVIEW

**Comment by amtoine** [comment]
appears like the CI is upset by these changes @ayax79 :wink:

**Comment by ayax79** [comment]
> appears like the CI is upset by these changes @ayax79 😉

On it!

**Comment by amtoine** [comment]
a simple API change, with a new flag that preserves the previous behaviour, sounds good @ayax79 :pray:

**Comment by amtoine** [comment]
@ayax79 
do you think this PR will solve https://github.com/nushell/nushell/issues/10123?
to know if it should be linked :relieved:

**Comment by ayax79** [comment]
> @ayax79 do you think this PR will solve #10123? to know if it should be linked 😌

Yes, this fixes it. It now builds against polars-arrow 0.32.1.

File: crates/nu-cmd-dataframe/src/dataframe/values/nu_expression/mod.rs [hunk 1]
```diff
@@ -433,5 +426,8 @@ pub fn expr_to_value(expr: &Expr, span: Span) -> Value {
                 span,
             )
         }
+        // Selector is part of the polars_plan package. It is currently marked as nightly.
+        // Wasn't sure what the best approach to take here would be.
+        Expr::Selector(_) => todo!("Add selector support"),
```
Comment (fdncred) [inline crates/nu-cmd-dataframe/src/dataframe/values/nu_expression/mod.rs:431]: I believe we should use some type of `ShellError::UnsupportedInput` or create a NonImpleletedYet versus having a `todo!`.

**Comment by fdncred** [comment]
Looks good @ayax79, we just shouldn't have `todo!. Looks like there's one in arg_true.rs that needs to be removed too at some point.

File: Cargo.lock [hunk 2]
```diff
@@ -4837,6 +4825,25 @@ dependencies = [
...
+
+[[package]]
+name = "sqlparser"
+version = "0.37.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "37ae05a8250b968a3f7db93155a84d68b2e6cea1583949af5ca5b5170c76c075"
+dependencies = [
+ "log",
+ "serde",
+]
```
Comment (sholderbach) [inline Cargo.lock:4845]: As apparently `polars 0.32.1` is still using `sqlparser 0.36.1`, could you use `0.36.1` as well to not duplicate the dependency? In the same vein `nu-command` still has a dependency on `sqlparser 0.34`.

https://github.com/nushell/nushell/blob/1f06f8405c8e2553574a8327944894722e147b9d/crates/nu-command/Cargo.toml#L83

Comment (ayax79) [reply Cargo.lock:4845]: Will do

File: crates/nu-cmd-dataframe/src/dataframe/values/nu_expression/mod.rs (hunk 1 above)
Comment (ayax79) [reply crates/nu-cmd-dataframe/src/dataframe/values/nu_expression/mod.rs:431]: I'll change.. I dug in a little more, my comment here is incorrect and updated it.  It looks like polars_plan::dsl::selector::Selector isn't publicly exposed, so we simply can't do anything with it directly. I'll dig through polars a bit more to see if there is anything I can do with the reference. If not I will change to UnsupportedInput.

**Comment by ayax79** [comment]
> Looks good @ayax79, we just shouldn't have `todo!`. Looks like there's one in arg_true.rs that needs to be removed too at some point.

I'll see if I can figure out what is going on with the second todo!

**Comment by ayax79** [comment]
@fdncred, @sholderbach  - This should be good to go now.

**Review by fdncred — APPROVED** [review]
Cool. I skimmed through these changes. You've addressed my concerns. Thanks!

**Review by sholderbach — APPROVED** [review]
Thanks for getting us up to date. I just updated the dependency on the unrelated `nu-command` side, then we should be good to go

**Pull request merged**